libavformat/mpegts.c

```diff
@@ -2175,8 +2181,19 @@ int ff_parse_mpeg2_descriptor(AVFormatContext *fc, AVStream *st, int stream_type
 
             st->codecpar->codec_type = AVMEDIA_TYPE_SUBTITLE;
             st->codecpar->codec_id   = AV_CODEC_ID_ARIB_CAPTION;
-            st->codecpar->profile    = picked_profile;
+            if (st->codecpar->profile != picked_profile) {
+                st->codecpar->profile = picked_profile;
+                sti->need_context_update = 1;
+            }
+            if (v_st &&
+                (st->codecpar->width != v_st->codecpar->width ||
+                 st->codecpar->height != v_st->codecpar->height)) {
+                st->codecpar->width = v_st->codecpar->width;
+                st->codecpar->height = v_st->codecpar->height;
+                sti->need_context_update = 1;
+            }
```

Comment:
This section isn't appropriate. Instead, provide an AVOption with type `AV_OPT_TYPE_IMAGE_SIZE` in your decoder, defaulting to the layout dimensions derived from the subtitle stream itself, and allow player apps to set it to an appropriate value (this is usually the _output_ resolution) when using the bitmap renderer.

Reply:
I'll consider it.

Reply:
I add `-canvas_size` option to specify input video size.
The same option already exists in `ffmpeg` tool, and it set `width` and `height` in `AVCodecContext`. Since `ffmpeg` tool doesn't pass `-canvas_size` option to `libaribcaption`, thus I use the values in `AVCodecContext`. The `-canvas_size` option in `AVOption` is used by other tools such as ffplay.

Note that the `canvas_size` specifies frame size of input video (i.e., SAR basis). Actual canvas size is stretched to 16:9 to match DAR-derived video size.